lib/grack/git.rb

```diff
@@ -0,0 +1,58 @@
+module Grack
+  class Git
+    attr_reader :repo
+    def initialize(git_path, repo_path)
+      @git_path = git_path
+      @repo = repo_path
+      @repo = nil unless valid_git_dir?
+    end
+
+    def update_server_info
+      execute(%W(update-server-info))
+    end
+
+    def command(cmd)
+      [@git_path || 'git'] + cmd
+    end
+
+    def capture(cmd)
+      IO.popen(popen_env, cmd, popen_options).read
+    end
+
+    def execute(cmd)
+      capture( command(cmd) ).chomp
+    end
+
+    def execute_with_block(cmd)
```

Comment:
How easy is it to move this to `execute` and check `block_given?` there? I prefer one `execute` method.